ksqldb-functional-tests/src/test/resources/query-validation-tests/multi-joins.json

```diff
@@ -813,6 +813,70 @@
         ]
       }
     },
+    {
+      "name": "stream-stream-stream - outer-outer",
+      "statements": [
+        "CREATE STREAM S1 (ID INT KEY, V0 bigint) WITH (kafka_topic='left', value_format='JSON');",
+        "CREATE STREAM S2 (ID INT KEY, V0 bigint) WITH (kafka_topic='right', value_format='JSON');",
+        "CREATE STREAM S3 (ID INT KEY, V0 bigint) WITH (kafka_topic='right2', value_format='JSON');",
+        "CREATE STREAM OUTPUT as SELECT ROWKEY, s1.V0, s2.V0, s3.V0 FROM S1 FULL OUTER JOIN S2 WITHIN 5 seconds ON S1.ID = S2.ID FULL OUTER JOIN S3 WITHIN 10 seconds ON S1.ID = S3.ID;"
+      ],
+      "inputs": [
+        {"topic": "left", "key": 0, "value": {"V0": 1}, "timestamp": 0},
+        {"topic": "right", "key": 0, "value": {"V0": 2}, "timestamp": 1},
+        {"topic": "right2", "key": 0, "value": {"V0": 3}, "timestamp": 2},
+        {"topic": "left", "key": 0, "value": {"V0": 4}, "timestamp": 3}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"S1_V0": 1, "S2_V0": null, "S3_V0": null}, "timestamp": 0},
+        {"topic": "OUTPUT", "key": 0, "value": {"S1_V0": 1, "S2_V0": 2, "S3_V0": null}, "timestamp": 1},
+        {"topic": "OUTPUT", "key": 0, "value": {"S1_V0": 1, "S2_V0": null, "S3_V0": 3}, "timestamp": 2},
```

Comment:
Same question as above (below).

Reply:
Right now with 1) @spena 's fix and 2) https://confluentinc.atlassian.net/browse/KSE-326?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel, I think this is reasonable: after we have piped `2` from `right`, on the second sub-topology we would have buffered two records: `(1, null)`, and `(1, 2)` so when `3` from `right2` is received, two outputs, i.e. line 3/4 would be emitted.

In practice though, things could be more tricky than the test driver, since there's a repartition topic between `(left join right)` and `right2`, when a record from `right2` is received, the previous join result `(1, null)` and `(1, 2)` may or may not have piped from the repartition topic, so we may also see a `(null, null, 3)` be emitted first, also maybe we would not emit `(1, 2, null)` if the intermediate record at the repartition topic was fetched a bit late: in that case the output ordering would be `(1, null, null)` then `(1, null, 3)` and then `(1, 2, 3)`, etc.

Once both 1) and 2) above are fixed, we should only see two outputs, which is `(1,2,3)` and `(4,2,3)`.

Reply:
> since there's a repartition topic between

Not necessarily. Also, this test should not use a repartition as we join on the input key on all 3 streams. But you are of course right, that in the most general case, there _could_ be a repartition topic.

Reply:
Thanks, both. I have filed https://github.com/confluentinc/ksql/issues/7623 to track the unnecessary repartition. (This is the same as KSE-326 mentioned above, for anyone following this thread.)